commit/factory.go

```diff
@@ -35,42 +38,88 @@ func (p PluginFactoryConstructor) NewReportingPluginFactory(
-type PluginFactory struct{}
+type PluginFactory struct {
+	lggr            logger.Logger
+	ocrConfig       reader.OCR3ConfigWithMeta
+	commitCodec     cciptypes.CommitPluginCodec
+	msgHasher       cciptypes.MessageHasher
+	homeChainReader reader.HomeChain
+	contractReaders map[cciptypes.ChainSelector]types.ContractReader
+	chainWriters    map[cciptypes.ChainSelector]types.ChainWriter
+}
 
-func NewPluginFactory() *PluginFactory {
-	return &PluginFactory{}
+func NewPluginFactory(
+	lggr logger.Logger,
+	ocrConfig reader.OCR3ConfigWithMeta,
+	commitCodec cciptypes.CommitPluginCodec,
+	msgHasher cciptypes.MessageHasher,
+	homeChainReader reader.HomeChain,
+	contractReaders map[cciptypes.ChainSelector]types.ContractReader,
+	chainWriters map[cciptypes.ChainSelector]types.ChainWriter,
+) *PluginFactory {
+	return &PluginFactory{
+		lggr:            lggr,
+		ocrConfig:       ocrConfig,
+		commitCodec:     commitCodec,
+		msgHasher:       msgHasher,
+		homeChainReader: homeChainReader,
+		contractReaders: contractReaders,
+		chainWriters:    chainWriters,
+	}
 }
 
-func (p PluginFactory) NewReportingPlugin(config ocr3types.ReportingPluginConfig,
+func (p *PluginFactory) NewReportingPlugin(config ocr3types.ReportingPluginConfig,
 ) (ocr3types.ReportingPlugin[[]byte], ocr3types.ReportingPluginInfo, error) {
-	// TODO: Get this from ocr config, it's the mapping of the oracleId index in the DON
-	var oracleIDToP2pID map[commontypes.OracleID]libocrtypes.PeerID
+	var oracleIDToP2pID = make(map[commontypes.OracleID]ragep2ptypes.PeerID)
+	for i, p2pID := range p.ocrConfig.Config.P2PIds {
+		oracleIDToP2pID[commontypes.OracleID(i)] = p2pID
+	}
+
 	onChainTokenPricesReader := reader.NewOnchainTokenPricesReader(
 		reader.TokenPriceConfig{ // TODO: Inject config
 			StaticPrices: map[ocr2types.Account]big.Int{},
 		},
 		nil, // TODO: Inject this
 	)
+	ccipReader := reader.NewCCIPChainReader(
+		p.lggr,
+		p.contractReaders,
+		p.chainWriters,
+		p.ocrConfig.Config.ChainSelector,
+	)
 	return NewPlugin(
-		context.Background(),
-		config.OracleID,
-		oracleIDToP2pID,
-		pluginconfig.CommitPluginConfig{},
-		nil, // ccipReader
-		onChainTokenPricesReader,
-		nil, // reportCodec
-		nil, // msgHasher
-		nil, // lggr
-		nil, // homeChain
-	), ocr3types.ReportingPluginInfo{}, nil
+			context.Background(),
+			config.OracleID,
+			oracleIDToP2pID,
+			pluginconfig.CommitPluginConfig{
+				DestChain:           p.ocrConfig.Config.ChainSelector,
+				NewMsgScanBatchSize: 256,
+			},
+			ccipReader,
+			onChainTokenPricesReader,
+			p.commitCodec,
+			p.msgHasher,
+			p.lggr,
+			p.homeChainReader,
+		), ocr3types.ReportingPluginInfo{
+			Name: "CCIPRoleCommit",
+			Limits: ocr3types.ReportingPluginLimits{
+				// No query for this commit implementation.
+				MaxQueryLength:       0,
+				MaxObservationLength: 20_000, // 20kB
+				MaxOutcomeLength:     10_000, // 10kB
+				MaxReportLength:      10_000, // 10kB
```

Comment:
These are ballpark, we'll revisit as we test.

Reply:
Will this be read from config in the future?

Reply:
No these will likely be constants, but might not be these values.